pkg/controller/nodeipam/node_ipam_controller.go

```diff
@@ -90,19 +91,19 @@ func NewNodeIpamController(
 	allocatorType ipam.CIDRAllocatorType) (*Controller, error) {
 
 	if kubeClient == nil {
-		klog.Fatalf("kubeClient is nil when starting Controller")
+		return nil, errors.New("kubeClient is nil when starting Controller")
 	}
 
 	// Cloud CIDR allocator does not rely on clusterCIDR or nodeCIDRMaskSize for allocation.
 	if allocatorType != ipam.CloudAllocatorType {
 		if len(clusterCIDRs) == 0 {
-			klog.Fatal("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
+			return nil, errors.New("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
```

Comment:
```suggestion
			return nil, fmt.Errorf("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
```

Reply:
Updated similar errors in this function as well.